### PR TITLE
Improvement/add mocha typescript nyc

### PR DESCRIPTION
**Pull request by bspindler**

### CODE REVIEW

**Comment by bspindler** [comment]
This PR is not ready.  It's incorrectly referencing `js/bin/test.js` file.  I based most of this on this seed project https://github.com/pana-cc/mocha-typescript-seed.
npm test doesn't pick up the one test.ts file

**Comment by bspindler** [comment]
OK this PR should be ready, lmk.

File: tsconfig.json [hunk 1]
```diff
@@ -8,13 +8,17 @@
...
     "suppressImplicitAnyIndexErrors": true,
     "outDir": "./js"
   },
   "types": [
       "node"
   ],
   "include": [
-    "./ts/**/*.ts"
+    "./ts/**/*.ts",
+    "./test/*.ts"
```
Comment (TheConnMan) [inline tsconfig.json]: Including tests here is what creates the `ts/` subfolder and forces `metricly.js` under another folder. You can run the tests in TypeScript, so you don't need to precompile them by updating the `mocha.opts` to point to the TS files. I personally think that's cleaner and you don't have to keep compiling the tests before testing. Gets rid of the `pretest` and `prepare` scripts in `package.json`, too.

File: package.json [hunk 2]
```diff
@@ -7,21 +7,28 @@
...
+    "watch": "mocha-typescript-watch"
   },
   "bin": {
-    "metricly": "js/bin/metricly.js"
+    "metricly": "js/ts/bin/metricly.js"
   },
   "author": "TheConnMan",
   "license": "ISC",
   "dependencies": {
+    "@types/chai": "^4.1.2",
```
Comment (TheConnMan) [inline package.json]: All `@types/`, mocha, chai, and nyc should be dev dependencies, not dependencies.

File: README.md [hunk 3]
```diff
@@ -28,3 +28,11 @@ node js/bin/metricly.js
 npm install -g ts-node
 ts-node ts/bin/metricly.ts
 ```
+
+## Testing
+// run tests
+```npm test```
```
Comment (TheConnMan) [inline README.md]: Should be `yarn test`. There isn't a functional difference, just keeping in line with using Yarn in this project.

**Comment by TheConnMan** [comment]
You should also add a test phase to the `.travis.yml` file.

**Comment by bspindler** [comment]
Thanks for the feedback!  I’ll take another pass a bit later today and clean this up.

File: tsconfig.json (hunk 1 above)
Comment (bspindler) [reply tsconfig.json]: Agreed. I think this takes care of it https://github.com/metricly/metricly-cli/pull/30/commits/ebbd0ae02a93bfff4fb7016c98bb084bc2c69d9f

**Pull request merged**